doc/integrator/upgrade_application.rst.mako

```diff
@@ -186,15 +186,38 @@ Layer definition for ngeo clients is separate and different from layer
 definition for CGXP clients, see :ref:`administrator_administrate_layers`
 for details.
 To migrate the layer definitions from the CGXP structure to the ngeo
-structure, you can use the script ``.build/venv/bin/themev1tov2``.
+structure, you can use the script themev1tov2.
+
+In non docker context, run the script from venv:
+
+.. prompt:: bash
+
+   .build/venv/bin/themev1tov2 -i geoportal/production.ini
+
+In docker context, run the script from geoportal service:
+
+.. prompt:: bash
+
+   docker-compose up -d
+   docker-compose run --rm geoportal themev1tov2 -i production.ini
```

Comment:
Why not:
docker-compose exec geoportal themev1tov2 --app-config=production.ini
?

Reply:
Note that I got an error for redis, but as I'm not sure db is not in composition is all cases, I've run the entire composition before themev1tov2 in documentation.

If I do not mistake, if links were configured in composition we could only do:
```
docker-compose run --rm geoportal themev1tov2 -i production.ini
```
As we not not need geoportal web service running.

Reply:
We are used to say that running multiples process in a docker container is not a good practice.
I do not know if docker-compose exec logs would appear in container logs ?

Reply:
- It's not running multiples process that's not a good practice, but running multiple services.
- In general I prefer `docker-compose exec` because the output is cleaner.
- But in this case we don't need to access to any other services? (the db will be in the docker-compose only for quick tests not for this kind of integration task) than we can use `docker-compose run` with `docker-compose up` is right?
- Where and what kind or redis error do you get?
- Neither of `run` and `exec` will interact with `docker-compose logs`...

Reply:
Do not forgot that we run a full web app in themev1tov2 to get the settings.
```
$ docker-compose run --rm geoportal themev1tov2
Creating network "apronamain_default" with the default driver
Evaluate: /app/aprona_geoportal/static/lib/cgxp/dygraphs/dygraphs/dygraphs/jsdoc-toolkit/templates/jsdoc/allfiles.tmpl
Evaluate: /app/aprona_geoportal/static/lib/cgxp/dygraphs/dygraphs/dygraphs/jsdoc-toolkit/templates/jsdoc/allclasses.tmpl
Evaluate: /app/aprona_geoportal/static/lib/cgxp/dygraphs/dygraphs/dygraphs/jsdoc-toolkit/templates/jsdoc/symbol.tmpl
Evaluate: /app/aprona_geoportal/static/lib/cgxp/dygraphs/dygraphs/dygraphs/jsdoc-toolkit/templates/jsdoc/class.tmpl
Evaluate: /app/aprona_geoportal/static/lib/cgxp/dygraphs/dygraphs/dygraphs/jsdoc-toolkit/templates/jsdoc/index.tmpl
Evaluate: /app/aprona_geoportal/static-ngeo/build/desktop.html.tmpl
Evaluate: /app/aprona_geoportal/static-ngeo/build/mobile.html.tmpl
Evaluate: /app/aprona_geoportal/static-ngeo/build/desktop.71a426.css.tmpl
Evaluate: /app/aprona_geoportal/static-ngeo/build/mobile.2b98f2.css.tmpl
Evaluate: /app/webpack.apps.js.tmpl
2018-11-09 14:11:28,240 INFO  [c2cwsgiutils.broadcast][140573415221056] Switching from a local broadcaster to a redis broadcaster
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/redis/connection.py", line 484, in connect
    sock = self._connect()
  File "/usr/local/lib/python3.6/dist-packages/redis/connection.py", line 511, in _connect
    socket.SOCK_STREAM):
  File "/usr/lib/python3.6/socket.py", line 745, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno -2] Name or service not known

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/redis/client.py", line 2408, in _execute
    return command(*args)
  File "/usr/local/lib/python3.6/dist-packages/redis/connection.py", line 610, in send_command
    self.send_packed_command(self.pack_command(*args))
  File "/usr/local/lib/python3.6/dist-packages/redis/connection.py", line 585, in send_packed_command
    self.connect()
  File "/usr/local/lib/python3.6/dist-packages/redis/connection.py", line 489, in connect
    raise ConnectionError(self._error_message(e))
redis.exceptions.ConnectionError: Error -2 connecting to redis:6379. Name or service not known.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/redis/connection.py", line 484, in connect
    sock = self._connect()
  File "/usr/local/lib/python3.6/dist-packages/redis/connection.py", line 511, in _connect
    socket.SOCK_STREAM):
  File "/usr/lib/python3.6/socket.py", line 745, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno -2] Name or service not known

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/themev1tov2", line 11, in <module>
    load_entry_point('c2cgeoportal-geoportal', 'console_scripts', 'themev1tov2')()
  File "/opt/c2cgeoportal_geoportal/c2cgeoportal_geoportal/scripts/themev1tov2.py", line 89, in main
    app = get_app(app_config, app_name, options=env)
  File "/usr/local/lib/python3.6/dist-packages/pyramid/paster.py", line 28, in get_app
    return loader.get_wsgi_app(name, options)
  File "/usr/local/lib/python3.6/dist-packages/plaster_pastedeploy/__init__.py", line 131, in get_wsgi_app
    global_conf=defaults)
  File "/usr/local/lib/python3.6/dist-packages/paste/deploy/loadwsgi.py", line 247, in loadapp
    return loadobj(APP, uri, name=name, **kw)
  File "/usr/local/lib/python3.6/dist-packages/paste/deploy/loadwsgi.py", line 272, in loadobj
    return context.create()
  File "/usr/local/lib/python3.6/dist-packages/paste/deploy/loadwsgi.py", line 710, in create
    return self.object_type.invoke(self)
  File "/usr/local/lib/python3.6/dist-packages/paste/deploy/loadwsgi.py", line 229, in invoke
    filtered = context.next_context.create()
  File "/usr/local/lib/python3.6/dist-packages/paste/deploy/loadwsgi.py", line 710, in create
    return self.object_type.invoke(self)
  File "/usr/local/lib/python3.6/dist-packages/paste/deploy/loadwsgi.py", line 146, in invoke
    return fix_call(context.object, context.global_conf, **context.local_conf)
  File "/usr/local/lib/python3.6/dist-packages/paste/deploy/util.py", line 55, in fix_call
    val = callable(*args, **kw)
  File "/app/aprona_geoportal/__init__.py", line 24, in main
    config.include('c2cgeoportal_geoportal')
  File "/usr/local/lib/python3.6/dist-packages/pyramid/config/__init__.py", line 839, in include
    c(configurator)
  File "/opt/c2cgeoportal_geoportal/c2cgeoportal_geoportal/__init__.py", line 517, in includeme
    config.include('c2cwsgiutils.pyramid.includeme')
  File "/usr/local/lib/python3.6/dist-packages/pyramid/config/__init__.py", line 839, in include
    c(configurator)
  File "/opt/c2cwsgiutils/c2cwsgiutils/pyramid.py", line 20, in includeme
    broadcast.init(config)
  File "/opt/c2cwsgiutils/c2cwsgiutils/broadcast/__init__.py", line 40, in init
    _broadcaster = redis.RedisBroadcaster(redis_url, broadcast_prefix)
  File "/opt/c2cwsgiutils/c2cwsgiutils/broadcast/redis.py", line 26, in __init__
    self._pub_sub.subscribe(**{self._get_channel('c2c_dummy'): lambda message: None})
  File "/usr/local/lib/python3.6/dist-packages/redis/client.py", line 2482, in subscribe
    ret_val = self.execute_command('SUBSCRIBE', *iterkeys(new_channels))
  File "/usr/local/lib/python3.6/dist-packages/redis/client.py", line 2404, in execute_command
    self._execute(connection, connection.send_command, *args)
  File "/usr/local/lib/python3.6/dist-packages/redis/client.py", line 2415, in _execute
    connection.connect()
  File "/usr/local/lib/python3.6/dist-packages/redis/connection.py", line 489, in connect
    raise ConnectionError(self._error_message(e))
redis.exceptions.ConnectionError: Error -2 connecting to redis:6379. Name or service not known.
```

Reply:
It might be possible to use get_appsettings and merge yaml config manually in themev1tov2 script, but not sure this is really necessary.

Reply:
We should add a way to don't include c2cwsgiutils.pyramid when we build the application from a script...

Reply:
I'm not sure that the get_appsettings is enough, but we can try it :-)

Reply:
Is it really needed to change something as this script works with composition up.
IMHO we should just merge doc update.